src/code/z_kankyo.c

```diff
@@ -732,13 +1694,124 @@ void func_800F8970(void) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdatePostmanEvents.s")
+void Environment_UpdatePostmanEvents(PlayState* play) {
+    u16 temp_a2_2;
+    u8 eventInfBits;
+    u8 day = (gSaveContext.eventInf[7] & 0xE0) >> 5;
```

Comment:
Should we make macros for these bits?

Reply:
We should yeah, but I was struggling with the `SET` macro for:
```c
eventInfBits = gSaveContext.eventInf[7] & (u8)~0xE0;
eventInfBits |= (u8)(day << 5);
gSaveContext.eventInf[7] = eventInfBits;
```

I was hoping `eventInfBits` could be a local temp, something like:
```c
{ \
    u8 __eventInfBits; \
    __eventInfBits = gSaveContext.eventInf[7] & (u8)~0xE0; \
    __eventInfBits |= (u8)(day << 5); \
    gSaveContext.eventInf[7] = __eventInfBits; \
}
```
then you could just define a macro that takes in a `day` argument, but then the scoping of this breaks matching.

Thoughts?

Reply:
z_schedule has a similar problem by not being able to declare the temp as part of the macro.
It works around it by taking the temp as a parameter of the macro. See `SCHEDULE_CALC_TIME`

Reply:
I see, I gave it a shot, let me know what you think